controllers/secret_controller.go

```diff
@@ -299,10 +308,17 @@ func syncSecrets(src, dest *corev1.Secret) {
 	}
 
 	for k, v := range src.Labels {
-		dest.Labels[k] = v
+		if !filterValue(k) {
```

Comment:
I was thinking about this being a bit more generic than just cass-operator's. We might have similar annotations / labels / requirements with reaper's secrets or medusa secrets also. Whether they are always datacenterspecific or something else I don't know yet. That's why the filter name. 

It could also just be something like "watched-by" label / annotation that some other controller (not even necessarily ours) uses that we might not want to copy.